### PR TITLE
Ops 2376

### PR DESCRIPTION
**Pull request by cytopia**
### Upcoming git tag: `v1.0.0`

### CODE REVIEW

File: tasks/create_nat_gateway.yml [hunk 1]
```diff
@@ -58,9 +65,49 @@
     - natgw.eip_name is defined
 
 
+# Idempotence Issue
+# -----------------
+#
+# ec2_vpc_nat_gateway returns a hard-coded ID in chek mode (nat-123456789)
```
Comment (danvaida) [inline tasks/create_nat_gateway.yml]: *check

File: tasks/create_nat_gateway.yml [hunk 2]
```diff
@@ -58,9 +65,49 @@
     - natgw.eip_name is defined
 
 
+# Idempotence Issue
```
Comment (danvaida) [inline tasks/create_nat_gateway.yml]: IMHO, this should be part of the README, in some sort of a "caveats" section.

File: tasks/create_nat_gateway.yml [hunk 3]
```diff
@@ -58,9 +65,49 @@
...
+# ec2_vpc_nat_gateway returns a hard-coded ID in chek mode (nat-123456789)
+# this results in showing the playbook as 'changed' when doing a dry run.
+# To overcome this issue we do the following:
+#
+# 1. Gather existing NAT GW in that subnet
+# 2. If found:
+#    2.1. Use its ID for tagging
+# 3. If not found:
+#    3.1. Create it
+#    3.2. Gather facts again to find its id
```
Comment (danvaida) [inline tasks/create_nat_gateway.yml]: IIRC, its ID is returned when it's created, so point 3.2 would not be required. am I right?

File: tasks/create_nat_gateway.yml (hunk 2 above)
Comment (cytopia) [reply tasks/create_nat_gateway.yml]: Issue has now been resolved by this alternative approach.

File: tasks/create_nat_gateway.yml (hunk 3 above)
Comment (cytopia) [reply tasks/create_nat_gateway.yml]: If not found and created in `check` mode, it still returns `nat-123456789` which will show the tagging task as changed.

Comment (danvaida) [reply tasks/create_nat_gateway.yml]: in that case, the module should not return bogus IDs when it detects that it's being ran in check mode. we should open an issue for this.

**Pull request merged**